tests/unit/_modules/test_packagemanager.py

```diff
@@ -0,0 +1,345 @@
+import pytest
+from srv.salt._modules.packagemanager import PackageManager, Zypper, Apt
+from mock import MagicMock, patch, mock
+
+
+class TestPackageManager():
+    '''
+    This class contains a set of functions that test srv.salt._modules.packagemanager
+    '''
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_opensuse(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('opensuse', 42.1, 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Zypper) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_suse(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('SUSE', '12.2', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Zypper) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_centos(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('centos', '8', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Apt) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_fedora(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('fedora', '23', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Apt) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    @mock.patch('srv.salt._modules.packagemanager.Popen')
+    def test_reboot(self, po, dist_return):
+        """
+        Reboot method
+        """
+        dist_return.return_value = ('opensuse', '42.2', 'x86_64')
+        ret = PackageManager()
+        ret.reboot_in()
+        assert po.called is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    @mock.patch('srv.salt._modules.packagemanager.Popen')
+    def test_not_implemented(self, po, dist_return):
+        """
+        Your platform is not supported
+        """
+        dist_return.return_value = ('ScientificLinux', '42.2', 'x86_64')
```

Comment:
So it's like an apt wrapper for zypper? Hmm ...

That's what it seems: https://software.opensuse.org/package/zypper-aptitude

My initial thoughts are that I think we still might be fine if the system reports apt for this case sense we should theoretically (in practice may be different) just use apt commands and they get sent to zypper. It's possible that detecting when a reboot is necessary via zypper-aptitude could not follow apt's normal procedures, though. It's probably not worth risking using apt when zypper is available. Thinking a little more, the order of the statements as I wrote them is such that if a system has zypper and zypper-aptitude, it will still report zypper since that check is first.

On the other hand ... a project called pacapt exists, and it uses distro to determine package manager. See code linked. Maybe this really is a better option?: https://github.com/icy/pacapt/blob/ng/lib/00_core.sh#L61
There is a comment in the code talking about Fedora 22 moving to dnf, which is also something we'd have to support.

More research: My dev machine has zypper-aptitude installed, and this is the apt-get version.
```sh
$> apt-get --version
zypper 1.13.21
```